### PR TITLE
Introduce configurable settings for NGINX static settings

### PR DESCRIPTION
**Pull request by akalipetis**
Now, the settings that can be changed are:

* DNS resolver
* Enabling/disabling the Let's Encrypt automatic SSL generation
* Fallback SSL certificate and key

The new settings are handled by templates, in combination with dockerize, which also waits for Redis to be accessible as a side-effect.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -9,21 +9,27 @@ services:
     depends_on:
       - redis
     environment:
+      CERYX_DISABLE_LE: ${CERYX_DISABLE_LE:-false}
```
Comment (parisk) [inline docker-compose.yml]: Can we rename this to `CERYX_DISABLE_LETS_ENCRYPT`?

File: ceryx/nginx/conf/ceryx.conf.tmpl [hunk 2]
```diff
@@ -14,11 +14,15 @@ server {
     listen 443 ssl;
     default_type text/html;
 
-    ssl_certificate /etc/ssl/resty-auto-ssl-fallback.crt;
-    ssl_certificate_key /etc/ssl/resty-auto-ssl-fallback.key;
+    ssl_certificate {{ default .Env.CERYX_SSL_CERT "/etc/ssl/resty-auto-ssl-fallback.crt" }};
+    ssl_certificate_key {{ default .Env.CERYX_SSL_CERT "/etc/ssl/resty-auto-ssl-fallback.key" }};
+
+    # Disable the LE certificate generation
+    {{ if ne (lower (default .Env.CERYX_DISABLE_LE "")) "true" }}
```
Comment (parisk) [inline ceryx/nginx/conf/ceryx.conf.tmpl]: `CERYX_DISABLE_LE` --> `CERYX_DISABLE_LETS_ENCRYPT` ?

File: README.md [hunk 3]
```diff
@@ -60,7 +71,7 @@ or open o pull request.
 
 ## Dynamic SSL certificates
 
-You can read more information on how to configure Ceryx with Dynamic SSL certificates [here](docs/dynamic-ssl).
+By default, Ceryx will try to generate a certificate when a domain is hit via HTTPS through Let's Encrypt, if and only if a route exists for it. If you don't want this to be enabled, you can use the `CERYX_DISABLE_LE` setting.
```
Comment (parisk) [inline README.md]: `CERYX_DISABLE_LE` --> `CERYX_DISABLE_LETS_ENCRYPT` ?

File: README.md [hunk 4]
```diff
@@ -20,17 +20,28 @@ a REST client.
...
-  * ``CERYX_NAME``: sets the API service name - defaults to ceryx
-  * ``CERYX_DEBUG``: enables debuging on the API service - defaults to true
   * ``CERYX_API_HOST``: sets the host that the API will bind to - defaults to 127.0.0.1
-  * ``CERYX_API_PORT``: sets the port that the API will listen - defaults to 5555
-  * ``CERYX_SERVER_NAME``: the URL of the API service - default to None
   * ``CERYX_API_HOSTNAME``: identical to `CERYX_SERVER_NAME`, but without imposing `Host` header limits - default to None
-  * ``CERYX_SECRET_KEY``: the path of the secret key to use - defaults to None
+  * ``CERYX_API_PORT``: sets the port that the API will listen - defaults to 5555
+  * ``CERYX_DEBUG``: enables debugging on the API service - defaults to true
+  * ``CERYX_DISABLE_LE``: if true, the automatic generation through Let's Encrypt does not kick in, defaults to false
```
Comment (parisk) [inline README.md]: `CERYX_DISABLE_LE` --> `CERYX_DISABLE_LETS_ENCRYPT` ?

**Comment by akalipetis** [comment]
@parisk did the rename, nice catch 👍

**Pull request merged**